package google.protobuf.serialize;

message MessageHolder {
    enum Action {
        CREATE = 0;
        RETRIEVE = 1;
        UPDATE = 2;
        DELETE = 3;
        CLOSE_CONNECTION = 4;
    }
    
    required Action action = 1 [default = CREATE];
    optional string type_name = 2;
    optional bytes body = 3;
    optional int32 return_value = 4 [default = 0];
}
